src/functions.php

```diff
@@ -25,15 +25,29 @@ function timeout(PromiseInterface $promise, $time, LoopInterface $loop)
             $loop->cancelTimer($timer);
             $reject($v);
         });
+    }, function ($resolveUnused, $reject) use ($loop, &$timer, $promise) {
+        // cancelling this promise will cancel the timer and reject
+        $loop->cancelTimer($timer);
+        $reject(new \RuntimeException('Timeout cancelled'));
```

Comment:
This could probably be removed, relying on the default cancelling behaviour of the input promise.